ls1intum/Artemis
# `Lecture`: Fix attachment unit file upload

### Checklist
#### General



#### Client


### Motivation and Context
Currently the fileupload to the server is somehow broken, when updating an lecture attachment unit with a file with the same name.
As hotfix for this issue this PR disables the `keepFilename` flag, so that unique filenames are generated. There will be a followup PR to make the `keepFilename` option work again and do some refactorings.

### Description


### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Lecture

1. Log in to Artemis
2. Navigate to Lecture Administration
3. Add a lecture unit attachment
4. Update the unit with a different file, but the same filename
5. verify that it is still possible to download this file.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
